### ISSUE TITLE
lib not being updated in jest tests

### ISSUE THREAD

**Issue by Rofer11607** [nrwl/nx#14320]
### Current Behavior

I have an internal buildable library that I have made changes to in my mono repo.
I build it using nx build crm-assets
I then try to use the library in the app.

one of the objects is clearly in the library (even typescript intelisense shows)
![image](https://user-images.githubusercontent.com/55964540/212166948-c78de0be-d8b4-43df-a50d-3af29c20704d.png)

![image](https://user-images.githubusercontent.com/55964540/212167117-2e2f13f1-6808-4266-aae8-b2ed86827aa2.png)

however the console output value at runtime is undefined. Even though the value itself is hardcoded
![image](https://user-images.githubusercontent.com/55964540/212167284-019880f8-3039-4fa2-b336-48787260c99e.png)

### Expected Behavior

The changes should reflect 

### Github Repo

https://github.com/Rofer11607/nxMinRepo

### Steps to Reproduce
go to apps/server/scheduled/crm-cache-sync/

then run crm-cache-sync.spec.ts

you should see that the cache for tableId 'bpytkg9tg' is undefined.
however if you go into libs/crm-assets/src/tables/payments

on line 4400 you should see it is indeed defined.

### Nx Report

```shell
Node : 16.16.0
   OS   : darwin arm64
   yarn : 1.22.17
   
   nx : 14.7.5
   @nrwl/angular : 14.7.5
   @nrwl/cypress : 14.7.5
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.7.5
   @nrwl/eslint-plugin-nx : 14.7.5
   @nrwl/express : Not Found
   @nrwl/jest : 14.7.5
   @nrwl/js : 14.7.5
   @nrwl/linter : 14.7.5
   @nrwl/nest : 14.8.6
   @nrwl/next : Not Found
   @nrwl/node : 14.7.5
   @nrwl/nx-cloud : 14.6.2
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 14.7.5
   @nrwl/web : 14.7.5
   @nrwl/workspace : 14.7.5
   typescript : 4.7.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
         ngx-spinner: 13.1.1
         @storybook/angular: 6.5.15
```

### Failure Logs

```shell
none
```

**Comment by FrozenPandaz** [comment]
Sorry, this is going to be hard to fix unless we have a reproduction. Try reaching out in the [nrwl community slack](https://go.nrwl.io/join-slack) to see if someone can help.

**Comment by Rofer11607** [comment]
Update, I can confirm that the issue only occurs in jest. When pulling the same lib through an api I get the correct value.

It seems that I made a change in the lib but jest is not getting the lastest change. 

I recently installed a package using  the flag --legacy-peer-deps, would this affect it?

**Comment by barbados-clemens** [comment]
maybe? it's hard to say without a reproducible project to look at.

**Comment by Rofer11607** [comment]
I went ahead and made a minimal repo.

to reproduce the issue you go to apps/server/scheduled/crm-cache-sync/

then run crm-cache-sync.spec.ts

you should see that the cache for tableId 'bpytkg9tg' is undefined.
however if you go into libs/crm-assets/src/tables/payments

on line 4400 you should see it is indeed defined.

**Comment by barbados-clemens** [comment]
Thanks for the reproduceable, but I'm having a hard time seeing what you're talking about. can you update the `crm-cache-sync.spec.ts` file to assert what you're expecting to happen so I can see the failing test. 

Also if you could cut down a little of the extra noise in the repo. the file you mentioned (`libs/crm-assets/src/tables/payments`) is a 6000+ line exported object. and you're doing extra filter/manipulation on the object. so I'm unsure what the actual issue is since it's hard to follow what is happening in the project.

**Comment by Rofer11607** [comment]
Thanks for the feedback! I went ahead and applied it.

the index of the lib now exports a small string and the test asserts the ideal case.

**Comment by barbados-clemens** [comment]
Thank you for updating the repo. 
looking into it the tests pass and correctly has the value from the library. 
all I've done is 
1. clone the repo
2. use nvm to use node 14, `nvm use 14`
3. install deps with yarn `yarn`
4. run tests `yarn nx test server`

can you share the error you're getting when you follow the same steps?

**Comment by Rofer11607** [comment]
Hmmm, that makes me think its an environment issue.

![image](https://user-images.githubusercontent.com/55964540/212416225-03fdfff9-3222-45b9-a37c-7289f8b876d3.png)

![image](https://user-images.githubusercontent.com/55964540/212416273-dbaed571-f519-4c80-9038-4e856fe7b976.png)


what could possibly cause this? I am encountering this on a fresh repo download as well as the original.

**Comment by Rofer11607** [comment]
same thing from nx server test
![image](https://user-images.githubusercontent.com/55964540/212417335-188a20fb-a442-4e5d-bafa-327ebc724493.png)

**Comment by barbados-clemens** [comment]
Unsure, but looks to be environmental at this point.
the repo you have in question is using yarn and yarn doesn't have --legacy-peer-deps flag as far as I'm aware (or don't see it in the docs?), that's an npm option though.

do you have jest/ts-node/nx globally installed? if so try removing those, I've seen issues with global versions causing strange issues.

**Comment by Rofer11607** [comment]
mmmm is it because I might have mixed npm and yarn? 

if I did npm i <package-x> --legacy-peer-deps would it mess the repo up? if so how do I fix?

**Comment by barbados-clemens** [comment]
delete node_modules.

if you want to use yarn then delete the package-lock.json file
if you want to use npm then delete the yarn.lock file

**Comment by Rofer11607** [comment]
Hey Guys,
I fixed it. The problem is that I ran nx migrate and it did not complete. The script is expecting a function at but receives an undefined value. I will post a separate issue to discuss.
